test/libsolidity/SemanticTest.cpp

```diff
@@ -151,8 +162,25 @@ map<string, Builtin> SemanticTest::makeBuiltins() const
 			[this](FunctionCall const& _call) -> optional<bytes>
 			{
 				soltestAssert(_call.arguments.parameters.empty(), "No arguments expected.");
-				  return toBigEndian(u256(storageEmpty(m_contractAddress) ? 1 : 0));
-		 	}
+				return toBigEndian(u256(storageEmpty(m_contractAddress) ? 1 : 0));
+			}
+		},
+	};
+}
+
+vector<SideEffectHook> SemanticTest::makeSideEffectHooks() const
+{
+	return {
+		[](FunctionCall const& _call) -> vector<string>
+		{
+			if (_call.signature.find("isoltest_side_effects") != string::npos)
```

Comment:
Since it's just for testing, I 'd keep this check as narrow as possible to reduce the chance of someone hitting it unintentionally:
```suggestion
			if (_call.signature == "isoltest_side_effects_test()")
```